src/backend/distributed/utils/maintenanced.c

```diff
@@ -257,6 +263,13 @@ CitusMaintenanceDaemonMain(Datum main_arg)
 		CHECK_FOR_INTERRUPTS();
 
 		/*
+		 * We clear the metadata cache before every iteration because otherwise
+		 * it might contain stale OIDs if deadlock detection runs during a DROP
+		 * EXTENSION followed by a CREATE EXTENSION.
+		 */
+		ClearMetadataCache();
```

Comment:
> However, pg_dist_partition is never opened in the background worker, we only call get_relname_relid to get the OID and then expect to later get an invalidation for that OID when the extension is dropped. That assumption doesn't hold, which means our cache gets stale.

Hm, I don't actually see the problem here. `CitusHasBeenLoaded()` intentionally does a `DistPartitionRelationId()` to make sure the oid is cached, `AvailableExtensionVersion()` makes sure caches are initialized. Relache callback are called regardless of the relation previously being opened in the current backend or not.  Hence it shouldn't matter whether `pg_dist_partition` has been opened in maintenanced beforehand.  What am I missing?